pkg/agent/baker.go

```diff
@@ -489,9 +488,7 @@ func getContainerServiceFuncMap(config *datamodel.NodeBootstrappingConfiguration
 			return profile.Distro == datamodel.CustomizedImage || profile.Distro == datamodel.CustomizedImageKata
 		},
 		"EnableHostsConfigAgent": func() bool {
-			return cs.Properties.OrchestratorProfile.KubernetesConfig != nil &&
-				cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster != nil &&
-				to.Bool(cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.EnableHostsConfigAgent)
```

Comment:
Could we set EnableHostsConfigAgent to false in RP to disable it and then remove related code in AgentBaker?

Reply:
Suggest to follow the steps to cleanup these code since we can quickly hotfix it if some customers are blocked by this change.